src/librustc_codegen_ironox/metadata.rs

```diff
@@ -31,8 +32,30 @@ impl MetadataLoader for IronOxMetadataLoader {
         Ok(rustc_erase_owner!(buf))
     }
 
-    fn get_dylib_metadata(&self, _target: &Target, filename: &Path)
+    fn get_dylib_metadata(&self, target: &Target, filename: &Path)
         -> Result<MetadataRef, String> {
-        unimplemented!("get_dylib_metadata {:?}", filename);
+        let mut input_file =
+            File::open(filename)
+                .map_err(|_e| format!("failed to read {}", filename.display()))?;
+        let mut buf = vec![];
+        input_file.read_to_end(&mut buf)
+            .map_err(|_| format!("failed to read {}", filename.display()))?;
+        let buf = OwningRef::new(box buf);
```

Comment:
What's `OwningRef` for?

Reply:
The metadata is supposed to be returned as a `MetadataRef`, which is defined as:
```
pub type MetadataRef = OwningRef<Box<dyn Erased>, [u8]>;
```
An `OwningRef` is a way of bundling an object (the 'owner') and a reference to some data it owns. For example, you might bundle a `vec` and a slice: `OwningRef<Vec<u8>, [u8]>`.

Reply:
OK.